Autopilot/PathManager/Inc/landingMode/landingMode.hpp

```diff
@@ -0,0 +1,113 @@
+    public:
+        /**
+         * Call to get singleton for this class
+         *
+         * @return instance of class object
+         */
+        static PathMode& getInstance();
+        
+        /**
+         * Execute the curent stage in the mode
+         * 
+         * @param telemetry_in -> telemetry data from ground 
+         * @param sensor_fusion_in -> sensor fusion data
+         * @param imu_data_in -> raw imu data
+         *
+         * @return none
+         */
+        void execute(Telemetry_PIGO_t telemetry_in, SFOutput_t sensor_fusion_in, IMU_Data_t imu_data_in);
+        
+        /**
+         * Get PathModeSelector singleton instance
+         *
+         * @return instance of PathModeSelector singleton
+         */
+        PathModeSelector* getModeSelector();
+
+        /**
+         * Returns a pointer for the current stage of flight
+         * 
+         * @return current stage of flight within mode
+         */
+        inline LandingModeStageManager* getCurrentStage() const { return current_stage; }
+        
+        /**
+         * Sets the pointer for the current stage of flight
+         * 
+         * @param new_stage -> new stage of flight within mode
+         * 
+         * @return none
+         */
+        inline void setCurrentStage(LandingModeStageManager* new_stage) { current_stage = new_stage; }
+
+        /**
+         * Returns a pointer for the current stage of flight
+         * 
+         * @return current stage of flight within mode
+         */
+        inline LandingStages getCurrentStageEnum() const { return landing_stage; }
+
+        /**
+         * Sets the enum for the current stage of flight
+         * 
+         * @param new_stage -> enum for new stage of flight within mode
+         * 
+         * @return none
+         */
+        inline void setCurrentStageEnum(LandingStages new_stage) { landing_stage = new_stage; }
+
+        /*
+         * Getters and setters for Landing Mode data
+         */
+        inline bool getMadeLandingPoints() { return made_landing_points; }
+
+        inline void setMadeLandingPoints(bool new_made_landing_points) { made_landing_points = new_made_landing_points; }
+
+        inline bool getIsPackage() { return is_package; }
+
+        inline void setIsPackage(bool new_is_package) { is_package = new_is_package; }
+
+        /**
+         * Resets control details for landing and takeoff stages
+         * 
+         * ResetPassby uses operator chaining, for more info, refer to the assignment operator chaining in the following link http://courses.cms.caltech.edu/cs11/material/cpp/donnie/cpp-ops.html
+         *
+         * @param control_details -> control details for current stage
+         * 
+         * @return none
+         */
+        void resetPassby(_PassbyControl* control_details){control_details->rollPassby = control_details->pitchPassby = control_details->throttlePassby = control_details->rudderPassby = false;}
```

Comment:
reset passby shouldn't only be in landing, it applies to all the control stages. There should have been a separate state within the previous PM state machine just for resetting this.